Automattic/wp-calypso
# Signup: Use v2 is-site-importable endpoint when importing

#### Changes proposed in this Pull Request

* Updates the `import` signup flow to use the v2 `is-site-importable` endpoint when checking site importability
* Displays appropriate error if url is not importable
    - Invalid or unresolveable urls
    - Sites that aren't powered by Wix or GoDaddy
    - Wix or GoDaddy sites with a `404` status (meaning there's no Wix or GoDaddy site at that url
    - Wix or GoDaddy sites with a site status other than `200` (meaning the url importer won't be able to import them
* Import signup UI is not changed (except for new error messages)

<img width="936" alt="Screen Shot 2019-08-06 at 13 08 24" src="https://user-images.githubusercontent.com/1699996/62564881-5040e900-b84b-11e9-9b7c-96ca5a9606f2.png">


#### Testing instructions



- Visit `/start/import` and complete the user step
- Verify error states on the url step
    - Enter an invalid url
    - Enter a site url not powered by Wix or GoDaddy
    - Enter a non-existant Wix or GoDaddy site (e.g. `foobarlalala.godaddysites.com`)
    - Enter an valid url that doesn't resolve (e.g. `foobarlalala.com`)
- Enter a valid Wix or GoDaddy url and continue
- You should be redirected to the Calypso import section and see a preview of the site to import
- Verify that you can successfully complete a Wix or GoDaddy import.
